src/screens/SignIn/index.tsx

```diff
@@ -13,20 +19,39 @@ export const signInScreenTestIds = {
   passwordField: 'sign-in-form__input-password',
   forgotButton: 'sign-in-form__forgot-button',
   signInButton: 'sign-in-form__button',
+  errorAlert: 'sign-in__error-alert',
+  loadingDialog: 'sign-in__loading-dialog',
 };
 
 const SignInScreen = (): JSX.Element => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const { loading, errors, success } = useAppSelector((state) => state.auth);
+
+  const navigate = useNavigate();
+
+  const dispatch = useAppDispatch();
+
   const handleSubmit = async (event: React.SyntheticEvent) => {
     event.preventDefault();
 
-    // TODO: call signIn
+    dispatch(signIn({ email, password }));
   };
 
+  useEffect(() => {
+    if (success) {
+      navigate(paths.dashboard, { replace: true });
+    }
+  }, [navigate, success]);
```

Comment:
do we need `navigate` here?

Reply:
@nvminhtue ah we can remove it 🙏

Reply:
@nvminhtue it seems we should not remove `navigate` since there will be a warning 
![image](https://github.com/manh-t/react-survey/assets/60863885/d0983498-5d42-4c90-bf51-6f1d8a2b5902)

Reply:
that's weird when we do not depend on `navigate` function. Alright, we can revert that 👍